web/packages/teleterm/src/ui/services/resources/resourcesService.ts

```diff
@@ -274,11 +279,12 @@ function makeGetResourcesParamsRequest(params: types.GetResourcesParams) {
   };
 }
 
-export type UnifiedResourceResponse =
+export type UnifiedResourceResponse = (
   | { kind: 'server'; resource: types.Server }
   | {
       kind: 'database';
       resource: types.Database;
     }
   | { kind: 'kube'; resource: types.Kube }
-  | { kind: 'app'; resource: types.App };
+  | { kind: 'app'; resource: types.App }
+) & { requiresRequest: boolean };
```

Comment:
It's kind of an unorthodox way to define a discriminated union. I'd say it might be better to repeat `requiresRequest` in every type or define some kind of `UnifiedResourceResponseBase<'app', types.App>`. Mostly because it'll result in a slightly more elegant type error.

Reply:
When I was adding it I felt that something is wrong, it's better to repeat the type.